Documentation/coding-guidelines/adding-api-guidelines.md

```diff
@@ -0,0 +1,124 @@
+      - If `TestTFMs` is empty it defaults to [netcoreapp1.0](https://github.com/dotnet/corefx/commit/57af41ef1439ad2e443e42d03d55d41613e4c02e#diff-cd0fc5e0bad8102e1a45aa7575bdd102R155)
+  - Add new test code following [conventions](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/project-guidelines.md#code-file-naming-conventions) for new files to that are specific to the new target framework.
+  - To run just the new test configuration run `msbuild <Library>.csproj /t:RebuildAndTest /p:TargetGroup=<TargetGroup>`
+
+##FAQ
+_**<a name="isnetstandard">Is your API part of netstandard?</a>**_
+
+- For netstandard2.0 refer to https://github.com/dotnet/standard/tree/master/netstandard/ref.
+- For earlier netstandard versions refer to list at https://github.com/dotnet/corefx/blob/master/pkg/NETStandard.Library/NETStandard.Library.packages.targets
+
+_**What is the difference between being part of netstandard and building against netstandard?**_
+
+Things that are part of netstandard can only change when we release a new version of a platform
+that supports the higher version of netstandard. Whereas things that build against netstandard and
+ship in independent packages can be changed without an update to the platform that it is running on.
+That gives more flexibility to add API to things that build against netstandard because it does not require
+a platform update to consume.
+
+_**What changes require me to update my target framework in my library package?**_
+
+*Target framework of dependencies*
+  - If any changes (ref or src) require a higher target framework of one of the dependencies then your library package target framework must be increased to match the new minimum version that supports all package dependencies.
+
+*Independently shipped dependencies*
+  - If any changes (ref or src) requires a newer runtime/platform change that ships independently(this should generally only apply to libraries that are part of netstandard). Some example dependencies include:
+    - Runtime (aka RID) split package dependencies
+      - Dependencies on runtime changes
+      - Dependencies on runtime libraries (i.e. in Microsoft.TargetingPack.Private.* packages)
+      - Dependencies on new native shim changes
+    - .NET Native Shared Library
+
+###Potential clean-up work that can be done
+- Remove old build configurations - The older build configurations will automatically be harvested from
+the last stable packages and thus can be removed.
+- Remove import statements - If not referencing any pre-netstandard stable packages the [imports of dotnet5.x](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/project.json#L28) are no longer needed and can be removed.
```

Comment:
These are all interesting for different reasons beyond adding new APIs. We should talk about them in our other guideline docs.